### PR TITLE
Make it pip-installable. Fix #10

### LINKED ISSUE #10: Make it pip installable
It might be a good idea to make it pip installable and move some of the "demos" to a separate repo. I can see how they grow, and how testing everything together might become an issue. Will you be interested in something like this?

### PR DESCRIPTION
**Pull request by Nagasaki45**

### CODE REVIEW

**Comment by Nagasaki45** [comment]
BTW, I don't mind maintaining the package on PyPI (mainly pushing new builds whenever they are ready).

**Comment by Nagasaki45** [comment]
Another issue, resulted from unpinning the versions, is some warnings:
```
deep_disfluency/rnn/lstm.py:112: UserWarning: DEPRECATION: If x is a vector, Softmax will not automatically pad x anymore in next releases. If you need it, please do it manually. The vector case is gonna be supported soon and the output will be a vector.
  y_t = T.nnet.softmax(T.dot(h_t, self.W_hy) + self.b_y)
```
I assume that changing our code to work properly with latest theano shouldn't be too hard.

```
UserWarning: Trying to unpickle estimator LogisticRegression from version 0.18.1 when using version 0.19.2. This might lead to breaking code or invalid results. Use at your own risk.
```
This one is more tricky. I see 4 options here, none of them is quite good:
- Sticking to 0.18.1. This is not how libraries should work.
- Assume that models are pickled fine, unpickle them with version each recent version of scikit-learn, and pickle again. Then, have some fancy way in our code to choose the correct models.
- Keep a more abstract info about the model, like a coefficients matrix, and load that into a model instead of unpickling a python object.
- Ignore :-)

**Comment by davidschlangen** [comment]
If I may barge in:

> `UserWarning: Trying to unpickle estimator LogisticRegression from version 0.18.1 when using version 0.19.2. This might lead to breaking code or invalid results. Use at your own risk.`

I think this is the most sensible solution (and the one recommended by the scikit learn people):

> * Keep a more abstract info about the model, like a coefficients matrix, and load that into a model instead of unpickling a python object.

**Comment by julianhough** [comment]
Many thanks Tom and David- yes, I think David's solution to getting round pickle problems is best. I will work on the abstract representation option for the Logistic Regression classifiers and scalers.

**Comment by Nagasaki45** [comment]
Sounds good!

Do you want to wait with merging until all of the issues are solved? Personally I'm already using it as a package by installing from my github repo branch. Will be happy to see this change in even with the warnings and other issues.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+    # This should be a valid link to your project's main homepage.
+    #
+    # This field corresponds to the "Home-Page" metadata field:
+    # https://packaging.python.org/specifications/core-metadata/#home-page-optional
+    url='https://github.com/dsg-bielefeld/deep_disfluency',  # Optional
+
+    # This should be your name or the name of the organization which owns the
+    # project.
+    author='TODO',  # Optional
```
[the remark this answers is not in the document]
Comment (julianhough) [reply setup.py]: Julian Hough, Tom Gurion, David Schlangen

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+    # This should be a valid email address corresponding to the author listed
+    # above.
+    author_email='TODO',  # Optional
```
[the remark this answers is not in the document]
Comment (julianhough) [reply setup.py]: julianchough@gmail.com

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    #     'dev': ['check-manifest'],
+    #     'test': ['coverage'],
+    # },
+
+    # If there are data files included in your packages that need to be
+    # installed, specify them here.
+    #
+    # If using Python 2.6 or earlier, then these have to be included in
+    # MANIFEST.in as well.
+    package_data={  # Optional
```
[the remark this answers is not in the document]
Comment (julianhough) [reply setup.py]: I think we only need the following sub-folders from data for the basic functionality:

deep_disfluency/data/tag_representations
deep_disfluency/data/lm_corpora

[the remark this answers is not in the document]
Comment (Nagasaki45) [reply setup.py]: Yeah, looks fine. Resolving.

**Comment by Nagasaki45** [comment]
@julianhough the changes we discussed are here.

**Comment by Nagasaki45** [comment]
@julianhough, please wait with adding `requirements.txt` again. There are ways to describe extra requirements for specific parts of a project. I think it might be more idiomatic solution. We can discuss that.

Thanks for merging!

**Comment by julianhough** [comment]
No probs.

Sure- I've put in a short-term solution for now with an internal requirements.txt in the experiments directory, but yes we could do it another way. This is part of a project on reproducing experimental results, so should probably keep the experimental part always up and running.

**Comment by Nagasaki45** [comment]
Yep, sounds good.

**Pull request merged**